rewrite-java-core/src/main/java/com/azure/recipes/v2/TypeReferenceRecipe.java

```diff
@@ -47,15 +47,30 @@ private static class ConvertTypeReferenceVisitor extends JavaIsoVisitor<Executio
                     .anyMatch(methodDeclaration -> methodDeclaration.getName().getSimpleName().equals("getRawType"));
 
             if (!alreadyTransformed && visitedNewClass.toString().contains("TypeReference")) {
```

Comment:
Same suggestion here the inferred type can be used here instead of string comparison to `TypeReference`.

Reply:
Addressed in latest commits.